### DISCUSSION TITLE
Is lsp-config too highly coupled with mason(-lspconfig)?

### DISCUSSION THREAD

**Discussion by MikaelElkiaer** [LazyVim/LazyVim#94] [General]
I noticed when trying to set up a lsp server, that could not be installed by mason, that it would not be set up.

It seems that this is due to the use of mason here: https://github.com/LazyVim/LazyVim/blob/99ee77a03d54553ce784da0e887fa8c1abeec865/lua/lazyvim/plugins/lsp/init.lua#L90

The `setup_handlers` method skips any lsp server for which it does not have a package: https://github.com/williamboman/mason-lspconfig.nvim/blob/a910b4d50f7a32d2f9057d636418a16843094b7c/lua/mason-lspconfig/init.lua#L40

So, the `servers` and `setup` are ignored for servers unrecognized by Mason.

This is a small nuisance, but it would be nice if these servers could be set up in a similar manner. :)

**Comment by folke** [comment]
That's a bug indeed. Will fix. 
See #95

**Comment by folke** [comment]
Should be fixed now.

LSP Servers not supported by mason, will now also be properly set up.

You can additionally specify `mason=false` if you want to install a server yourself without mason.

**Comment by MikaelElkiaer** [comment]
You are the best. :)
